### ISSUE TITLE
[Bug] Mapbox 3d terrain causing errors after upgrade from 8.8 to 8.9

### ISSUE THREAD

**Issue by fabiankaestner** [visgl/deck.gl#7742]
### Description

I'm developing an application using a MapboxOverlay layer to integrate deck.gl into a mapbox-gl-js map using Mapbox's 3d terrain (used in react with react-map-gl). This worked in 8.8, but with an elevation mismatch between deck.gl and Mapbox (see #7064). A partial fix has apparently been merged in #7114, which should now be in 8.9.
Upgrading the dependencies to deck.gl 8.9 causes the following error to be thrown repeatedly (seemingly on every rerender):
```
Uncaught TypeError: Cannot read properties of null (reading '0')
    at transformMat4 (vec4.js:486:13)
    at transformVector (math-utils.ts:10:18)
    at pixelsToWorld (web-mercator-utils.ts:456:18)
    at WebMercatorViewport.unproject (viewport.ts:290:19)
    at getBounds (get-bounds.ts:17:22)
    at WebMercatorViewport.getBounds (web-mercator-viewport.ts:291:21)
    at get subViewports [as subViewports] (web-mercator-viewport.ts:226:27)
    at DrawLayersPass._drawLayers (layers-pass.ts:97:37)
    at DrawLayersPass.render (layers-pass.ts:61:17)
    at DeckRenderer.renderLayers (deck-renderer.ts:78:35)
    at Deck._drawLayers (deck.ts:987:24)
    at Deck.redraw (deck.ts:539:12)
    at Map.<anonymous> (mapbox-overlay.ts:190:12)
    at Map.fire (evented.js:129:26)
    at Mapbox._fireEvent (mapbox.ts:881:14)
    at Map._render (map.js:3191:14)
    at map._render (mapbox.ts:557:17)
    at map.js:3559:26
```
Rendering performance is essentially unusable.

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [X] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

The error shouldn't occur.

### Steps to Reproduce

I've been able to reliably get a working map in 8.8 and reproduce the error in 8.9 with the following code:
```typescript
import React from "react";
import Map, { NavigationControl, Source, useControl } from "react-map-gl";
import { ScatterplotLayer } from "@deck.gl/layers";
import { MapboxOverlay } from "@deck.gl/mapbox";
import { MapboxOverlayProps } from "@deck.gl/mapbox/typed";

function DeckGLOverlay(
  props: MapboxOverlayProps & {
    interleaved?: boolean;
  }
) {
  const overlay = useControl<MapboxOverlay>(() => new MapboxOverlay(props));
  overlay.setProps(props);
  return null;
}

export function App() {
  const scatterplotLayer = new ScatterplotLayer({
    id: "my-scatterplot",
    data: [{ position: [-74.5, 40], size: 100 }],
    getPosition: (d) => d.position,
    getRadius: (d) => d.size,
    getFillColor: [255, 0, 0],
  });

  return (
    <Map
      initialViewState={{
        latitude: 40,
        longitude: -74.5,
        zoom: 12,
      }}
      style={{ width: "100vw", height: "100vh" }}
      terrain={{ source: "mapbox-dem", exaggeration: 1 }}
      mapStyle="mapbox://styles/mapbox/light-v9"
      mapboxAccessToken=""
    >
      <DeckGLOverlay layers={[scatterplotLayer]} />
      <Source
        id="mapbox-dem"
        type="raster-dem"
        url="mapbox://mapbox.mapbox-terrain-dem-v1"
        tileSize={512}
        maxzoom={14}
      />
      <NavigationControl />
    </Map>
  );
}

```

### Environment

- Framework version: deck.gl@8.9.1
- Browser: Chrome 110.0
- OS: MacOS 13.0.1 on M1 Max

### Logs

The error:
```
Uncaught TypeError: Cannot read properties of null (reading '0')
    at transformMat4 (vec4.js:486:13)
    at transformVector (math-utils.ts:10:18)
    at pixelsToWorld (web-mercator-utils.ts:456:18)
    at WebMercatorViewport.unproject (viewport.ts:290:19)
    at getBounds (get-bounds.ts:17:22)
    at WebMercatorViewport.getBounds (web-mercator-viewport.ts:291:21)
    at get subViewports [as subViewports] (web-mercator-viewport.ts:226:27)
    at DrawLayersPass._drawLayers (layers-pass.ts:97:37)
    at DrawLayersPass.render (layers-pass.ts:61:17)
    at DeckRenderer.renderLayers (deck-renderer.ts:78:35)
    at Deck._drawLayers (deck.ts:987:24)
    at Deck.redraw (deck.ts:539:12)
    at Map.<anonymous> (mapbox-overlay.ts:190:12)
    at Map.fire (evented.js:129:26)
    at Mapbox._fireEvent (mapbox.ts:881:14)
    at Map._render (map.js:3191:14)
    at map._render (mapbox.ts:557:17)
    at map.js:3559:26
```

Additionally, at the beginning of the log there's a warning from deck.gl:
```
deck: Pixel project matrix not invertible
```

**Comment by yuki0308** [comment]
I am also getting the same error, although not in React, when I view the 3d model from above.
I didn't get this error when I wasn't using mapbox-terrain-dem-v1.

[https://codesandbox.io/s/deck-gl-mapbox-overlay-rrhhw0](https://codesandbox.io/s/deck-gl-mapbox-overlay-rrhhw0)

[deckgl-mapbox.webm](https://user-images.githubusercontent.com/8759255/224908750-aa06bb45-708b-49d5-ace6-8d4d1b5bd5d9.webm)

**Comment by fabiankaestner** [comment]
I've made a codesandbox that reproduces this for me: https://codesandbox.io/s/practical-panini-i2m3kr

**Comment by bolatuly** [comment]
experiencing same issue after upgrading to 8.9.2

**Comment by Pessimistress** [comment]
Fixed in v8.9.3.
